libheif/plugins/encoder_aom.cc

```diff
@@ -237,7 +238,12 @@ static void aom_init_parameters()
   p->version = 2;
   p->name = kParam_threads;
   p->type = heif_encoder_parameter_type_integer;
-  p->integer.default_value = 4;
+  unsigned int threads = std::thread::hardware_concurrency();
+  if (threads == 0) {
+    // Could not autodetect, use previous default value.
+    threads = 4;
+  }
+  p->integer.default_value = threads;
```

Comment:
We should make sure `p->integer.default_value <= p->integer.maximum`. A computer may have more than 64 processor cores.

Reply:
Thank you, good catch. I have corrected this.

Reply:
Great! The new code looks good to me.